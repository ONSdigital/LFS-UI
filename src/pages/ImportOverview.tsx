import React, {Component} from "react";
import DocumentTitle from "react-document-title";
import {ReferenceFileImportTable} from "./manageBatch/ReferenceFileImportTable";

interface Props {
    match: any
}

interface State {

}

export class ImportOverview extends Component <Props, State> {
    displayName = ImportOverview.name;

    render() {
        return (
            <DocumentTitle title={"LFS Imports Overview"}>
                <div className="container">
                    <h3>Import Overview</h3>
                    <ReferenceFileImportTable/>
                </div>
            </DocumentTitle>
        );
    }
}