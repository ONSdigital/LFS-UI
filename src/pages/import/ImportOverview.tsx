import React, {Component} from "react";
import DocumentTitle from "react-document-title";
import {ImportFileTable} from "./ImportFileTable";

interface Props {
    match: any
}

interface State {

}

export class ImportOverview extends Component <Props, State> {
    displayName = ImportOverview.name;

    render() {
        return (
            <DocumentTitle title={"LFS Imports Overview"}>
                <div className="container">
                    <h3>Import Overview</h3>
                    <ImportFileTable/>
                </div>
            </DocumentTitle>
        );
    }
}