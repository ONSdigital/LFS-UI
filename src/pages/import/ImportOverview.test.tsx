import React from "react";
import {cleanup, fireEvent, render} from "@testing-library/react";

import {BrowserRouter} from "react-router-dom";

import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {ImportOverview} from "./ImportOverview";
import flushPromises from "../../tests/util/flushPromises";
import fetch from "../../tests/setup/__mocks__/fetch";
import MockDate from "mockdate";

// @ts-ignore
global.fetch = fetch;
MockDate.set('2015-01-21');

describe("Import Overview Page", () => {
    Enzyme.configure({adapter: new Adapter()});

    beforeEach(() => {
        // @ts-ignore
        process.env.NODE_ENV = "test";
    });

    afterEach(() => {
        cleanup();
    });


    function wrapper(render: any) {
        return render(
            <BrowserRouter>
                <ImportOverview match={""}/>
            </BrowserRouter>
        );
    }

    it("renders the page", async () => {
        expect(wrapper(render)).toMatchSnapshot();
    });

    it("should render without crashing", async () => {
        const {queryAllByTestId, getByText} = wrapper(render);
    });

    it("should display the page name in the Breadcrumbs", async () => {
        const {getByTestId} = wrapper(render);

        await flushPromises();

        let breadcrumbItems = getByTestId(/breadcrumbs/i);

        expect(breadcrumbItems.textContent).toEqual("Import Overview");
    });

    it("should display the list of all imports", async () => {
        const {getByTestId, queryAllByTestId, getByText} = wrapper(render);

        await flushPromises();

        let list = queryAllByTestId(/table-row/i);

        expect(list.length).toEqual(7);

        expect(getByText(/APS Design Weights/i)).toBeDefined();
        expect(getByText(/Bulk Amendments/i)).toBeDefined();
        expect(getByText(/Geographical Classifications/i)).toBeDefined();
        expect(getByText(/Output Specification/i)).toBeDefined();
        expect(getByText(/Population Estimates/i)).toBeDefined();
        expect(getByText(/Value Labels/i)).toBeDefined();
        expect(getByText(/Variable Definitions/i)).toBeDefined();
    });

    // TODO: Can't get Import link to change the pathname, despite this working on the Home page tests
    test.skip("the import buttons should redirect the page to the correct Import page", async () => {
        const {getByTestId, queryAllByTestId, getByText} = wrapper(render);

        await flushPromises();

        console.log(window.location.pathname);

        // delete window.location;
        // // @ts-ignore
        // window.location = {pathname: "import/overview"};

        console.log(window.location.pathname);

        let list = queryAllByTestId(/table-row/i);

        let importButton = list[0].children[3];

        console.log(importButton);

        await fireEvent.click(importButton);

        // await flushPromises();

        expect(window.location.pathname).toEqual("import/APS Design Weights");
    });
});