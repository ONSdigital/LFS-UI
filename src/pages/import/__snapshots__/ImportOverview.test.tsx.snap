// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import Overview Page renders the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <nav
          aria-label="Breadcrumb"
          class="breadcrumb"
          data-testid="breadcrumbs"
        >
          <ol
            class="breadcrumb__items"
          >
            <li
              class="breadcrumb__item u-fs-s breadcrumb__item--current"
              data-testid="breadcrumb-current"
              id="breadcrumb-1"
            >
              Import Overview
            </li>
          </ol>
        </nav>
        <h3>
          Import Overview
        </h3>
        <table
          class="table table--sortable"
          data-aria-asc="ascending"
          data-aria-desc="descending"
          data-aria-sort="Sort by"
          id="basic-table"
        >
          <thead
            class="table__head"
          >
            <tr
              class="table__row"
            >
              <th
                aria-sort="none"
                class="table__header "
                scope="col"
              >
                <button
                  aria-label="Sort by File Name"
                  class="table__sort-button"
                  data-index="0"
                  type="button"
                >
                  File Name
                </button>
              </th>
              <th
                aria-sort="none"
                class="table__header "
                scope="col"
              >
                <button
                  aria-label="Sort by Import Date"
                  class="table__sort-button"
                  data-index="1"
                  type="button"
                >
                  Import Date
                </button>
              </th>
              <th
                aria-sort="none"
                class="table__header "
                scope="col"
              >
                <button
                  aria-label="Sort by Status"
                  class="table__sort-button"
                  data-index="2"
                  type="button"
                >
                  Status
                </button>
              </th>
              <th
                class="table__header "
                scope="col"
              />
            </tr>
          </thead>
          <tbody
            class="table__body "
          >
            <tr
              class="table__row nonSelectableTableRow"
              data-testid="table-row"
              tabindex="0"
              title="Table Row"
            >
              <td
                class="table__cell "
              >
                APS Design Weights
              </td>
              <td
                class="table__cell "
              >
                04/02/2020
              </td>
              <td
                class="table__cell "
              >
                <span
                  class="status status--info"
                >
                  Imported
                </span>
              </td>
              <td
                class="table__cell "
              >
                <a
                  class="breadcrumb__link"
                  href="/import/APS Design Weights"
                >
                  Import
                </a>
              </td>
            </tr>
            <tr
              class="table__row nonSelectableTableRow"
              data-testid="table-row"
              tabindex="0"
              title="Table Row"
            >
              <td
                class="table__cell "
              >
                Bulk Amendments
              </td>
              <td
                class="table__cell "
              >
                04/02/2020
              </td>
              <td
                class="table__cell "
              >
                <span
                  class="status status--info"
                >
                  Not Imported
                </span>
              </td>
              <td
                class="table__cell "
              >
                <a
                  class="breadcrumb__link"
                  href="/import/Bulk Amendments"
                >
                  Import
                </a>
              </td>
            </tr>
            <tr
              class="table__row nonSelectableTableRow"
              data-testid="table-row"
              tabindex="0"
              title="Table Row"
            >
              <td
                class="table__cell "
              >
                Geographical Classifications
              </td>
              <td
                class="table__cell "
              >
                04/02/2020
              </td>
              <td
                class="table__cell "
              >
                <span
                  class="status status--info"
                >
                  File Older than One year
                </span>
              </td>
              <td
                class="table__cell "
              >
                <a
                  class="breadcrumb__link"
                  href="/import/Geographical Classifications"
                >
                  Import
                </a>
              </td>
            </tr>
            <tr
              class="table__row nonSelectableTableRow"
              data-testid="table-row"
              tabindex="0"
              title="Table Row"
            >
              <td
                class="table__cell "
              >
                Output Specification
              </td>
              <td
                class="table__cell "
              >
                04/02/2020
              </td>
              <td
                class="table__cell "
              >
                <span
                  class="status status--info"
                >
                  Not Imported
                </span>
              </td>
              <td
                class="table__cell "
              >
                <a
                  class="breadcrumb__link"
                  href="/import/Output"
                >
                  Import
                </a>
              </td>
            </tr>
            <tr
              class="table__row nonSelectableTableRow"
              data-testid="table-row"
              tabindex="0"
              title="Table Row"
            >
              <td
                class="table__cell "
              >
                Population Estimates
              </td>
              <td
                class="table__cell "
              >
                04/02/2020
              </td>
              <td
                class="table__cell "
              >
                <span
                  class="status status--info"
                >
                  Not Imported
                </span>
              </td>
              <td
                class="table__cell "
              >
                <a
                  class="breadcrumb__link"
                  href="/import/Population Estimates"
                >
                  Import
                </a>
              </td>
            </tr>
            <tr
              class="table__row nonSelectableTableRow"
              data-testid="table-row"
              tabindex="0"
              title="Table Row"
            >
              <td
                class="table__cell "
              >
                Value Labels
              </td>
              <td
                class="table__cell "
              >
                04/02/2020
              </td>
              <td
                class="table__cell "
              >
                <span
                  class="status status--info"
                >
                  Imported
                </span>
              </td>
              <td
                class="table__cell "
              >
                <a
                  class="breadcrumb__link"
                  href="/import/Value Labels"
                >
                  Import
                </a>
              </td>
            </tr>
            <tr
              class="table__row nonSelectableTableRow"
              data-testid="table-row"
              tabindex="0"
              title="Table Row"
            >
              <td
                class="table__cell "
              >
                Variable Definitions
              </td>
              <td
                class="table__cell "
              >
                04/02/2020
              </td>
              <td
                class="table__cell "
              >
                <span
                  class="status status--info"
                >
                  Imported
                </span>
              </td>
              <td
                class="table__cell "
              >
                <a
                  class="breadcrumb__link"
                  href="/import/Variable Definitions"
                >
                  Import
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <nav
        aria-label="Breadcrumb"
        class="breadcrumb"
        data-testid="breadcrumbs"
      >
        <ol
          class="breadcrumb__items"
        >
          <li
            class="breadcrumb__item u-fs-s breadcrumb__item--current"
            data-testid="breadcrumb-current"
            id="breadcrumb-1"
          >
            Import Overview
          </li>
        </ol>
      </nav>
      <h3>
        Import Overview
      </h3>
      <table
        class="table table--sortable"
        data-aria-asc="ascending"
        data-aria-desc="descending"
        data-aria-sort="Sort by"
        id="basic-table"
      >
        <thead
          class="table__head"
        >
          <tr
            class="table__row"
          >
            <th
              aria-sort="none"
              class="table__header "
              scope="col"
            >
              <button
                aria-label="Sort by File Name"
                class="table__sort-button"
                data-index="0"
                type="button"
              >
                File Name
              </button>
            </th>
            <th
              aria-sort="none"
              class="table__header "
              scope="col"
            >
              <button
                aria-label="Sort by Import Date"
                class="table__sort-button"
                data-index="1"
                type="button"
              >
                Import Date
              </button>
            </th>
            <th
              aria-sort="none"
              class="table__header "
              scope="col"
            >
              <button
                aria-label="Sort by Status"
                class="table__sort-button"
                data-index="2"
                type="button"
              >
                Status
              </button>
            </th>
            <th
              class="table__header "
              scope="col"
            />
          </tr>
        </thead>
        <tbody
          class="table__body "
        >
          <tr
            class="table__row nonSelectableTableRow"
            data-testid="table-row"
            tabindex="0"
            title="Table Row"
          >
            <td
              class="table__cell "
            >
              APS Design Weights
            </td>
            <td
              class="table__cell "
            >
              04/02/2020
            </td>
            <td
              class="table__cell "
            >
              <span
                class="status status--info"
              >
                Imported
              </span>
            </td>
            <td
              class="table__cell "
            >
              <a
                class="breadcrumb__link"
                href="/import/APS Design Weights"
              >
                Import
              </a>
            </td>
          </tr>
          <tr
            class="table__row nonSelectableTableRow"
            data-testid="table-row"
            tabindex="0"
            title="Table Row"
          >
            <td
              class="table__cell "
            >
              Bulk Amendments
            </td>
            <td
              class="table__cell "
            >
              04/02/2020
            </td>
            <td
              class="table__cell "
            >
              <span
                class="status status--info"
              >
                Not Imported
              </span>
            </td>
            <td
              class="table__cell "
            >
              <a
                class="breadcrumb__link"
                href="/import/Bulk Amendments"
              >
                Import
              </a>
            </td>
          </tr>
          <tr
            class="table__row nonSelectableTableRow"
            data-testid="table-row"
            tabindex="0"
            title="Table Row"
          >
            <td
              class="table__cell "
            >
              Geographical Classifications
            </td>
            <td
              class="table__cell "
            >
              04/02/2020
            </td>
            <td
              class="table__cell "
            >
              <span
                class="status status--info"
              >
                File Older than One year
              </span>
            </td>
            <td
              class="table__cell "
            >
              <a
                class="breadcrumb__link"
                href="/import/Geographical Classifications"
              >
                Import
              </a>
            </td>
          </tr>
          <tr
            class="table__row nonSelectableTableRow"
            data-testid="table-row"
            tabindex="0"
            title="Table Row"
          >
            <td
              class="table__cell "
            >
              Output Specification
            </td>
            <td
              class="table__cell "
            >
              04/02/2020
            </td>
            <td
              class="table__cell "
            >
              <span
                class="status status--info"
              >
                Not Imported
              </span>
            </td>
            <td
              class="table__cell "
            >
              <a
                class="breadcrumb__link"
                href="/import/Output"
              >
                Import
              </a>
            </td>
          </tr>
          <tr
            class="table__row nonSelectableTableRow"
            data-testid="table-row"
            tabindex="0"
            title="Table Row"
          >
            <td
              class="table__cell "
            >
              Population Estimates
            </td>
            <td
              class="table__cell "
            >
              04/02/2020
            </td>
            <td
              class="table__cell "
            >
              <span
                class="status status--info"
              >
                Not Imported
              </span>
            </td>
            <td
              class="table__cell "
            >
              <a
                class="breadcrumb__link"
                href="/import/Population Estimates"
              >
                Import
              </a>
            </td>
          </tr>
          <tr
            class="table__row nonSelectableTableRow"
            data-testid="table-row"
            tabindex="0"
            title="Table Row"
          >
            <td
              class="table__cell "
            >
              Value Labels
            </td>
            <td
              class="table__cell "
            >
              04/02/2020
            </td>
            <td
              class="table__cell "
            >
              <span
                class="status status--info"
              >
                Imported
              </span>
            </td>
            <td
              class="table__cell "
            >
              <a
                class="breadcrumb__link"
                href="/import/Value Labels"
              >
                Import
              </a>
            </td>
          </tr>
          <tr
            class="table__row nonSelectableTableRow"
            data-testid="table-row"
            tabindex="0"
            title="Table Row"
          >
            <td
              class="table__cell "
            >
              Variable Definitions
            </td>
            <td
              class="table__cell "
            >
              04/02/2020
            </td>
            <td
              class="table__cell "
            >
              <span
                class="status status--info"
              >
                Imported
              </span>
            </td>
            <td
              class="table__cell "
            >
              <a
                class="breadcrumb__link"
                href="/import/Variable Definitions"
              >
                Import
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
